src/check/parse/IR.zig

```diff
@@ -23,6 +23,7 @@ errors: []const Diagnostic,
 pub fn deinit(self: *IR) void {
     defer self.tokens.deinit();
     defer self.store.deinit();
+    self.store.gpa.free(self.errors);
```

Comment:
This fixed our memory leak... not sure it's 100% correct

Reply:
Probably correct for now, but might be wrong in the long term. Long term, we may want errors to outlive the rest of parse. So just like tokenize has finishAndDeinit, we may need something similar that returns the errors for later use but frees everything else. Not a now problem, but just context.

Reply:
Yeah, now you mention it... the errors should be pushed into the ModuleEnv as problems, not in the IR.

Reply:
Yeah, I'd rather not do this in deinit, errors should be owned by the owner of the parse result

Reply:
Can I leave this for you @gamebox to change in a follow up? I've left it here as it keeps the fuzzer happy.

Reply:
I thought the PR moved this into ModuleEnv?

Reply:
Also, the more proper short term fix would be to pull this out of deinit and instead add it next to all current callsites of `parse_ast.deint()`. That gives the caller control of when to free the error list.

Reply:
I'd like to leave this for a follow up, where we focus on unifying the approach to handling Problems across the compiler.